sensors/triggers/email/email.go

```diff
@@ -126,24 +126,34 @@ func (t *EmailTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 		return nil, fmt.Errorf("subject can't be empty")
 	}
 	t.Logger.Infow("sending emails...", zap.Any("to", emailTrigger.To))
-
-	notification := notifications.Notification{
-		Message: emailTrigger.Body,
-		Email: &notifications.EmailNotification{
-			Subject: emailTrigger.Subject,
-			Body:    emailTrigger.Body,
-		},
-	}
-	destination := notifications.Destination{
-		Service:   "email",
-		Recipient: emailTrigger.To,
+	var errs error
+	validEmail := regexp.MustCompile(`^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$`)
+	for _, addr := range emailTrigger.To {
+		if !validEmail.MatchString(addr) {
```

Comment:
Shall we move it to validate.go?

Reply:
As I understand validate.go does the pre validations when a sensor is created.

To support dynamic recipient address the validation was moved here. So that validation for to addresses will take place during execution.

Please correct me if I am wrong.

Reply:
@gokulav137 - you are right, missed that!